docs/docs/pages/testing-guides/tool-calling.mdx

```diff
@@ -0,0 +1,220 @@
+      ],
+      script: [
+        scenario.user(),
+        scenario.agent(),
+        (state) => expect(state.hasToolCall("get_current_weather")).toBe(true),
+        scenario.succeed(),
+      ], 
+    });
+    expect(result.success).toBe(true);
+  });
+});
+```
+:::
+
+## Asserting Tool Call Arguments
+
+Sometimes you need to check not just that a tool was called, but that it was called with the correct arguments.
+
+:::code-group
+```python [python]
+def check_tool_call_args(state):
+    tool_calls = state.latest_agent_message().tool_calls
+    assert tool_calls, "No tool calls found"
+    assert tool_calls[0].function.name == "get_current_weather"
+    assert "Paris" in tool_calls[0].function.arguments
+
+result = await scenario.run(
+    ...,
+    script=[
+        scenario.user("What's the weather in Paris?"),
+        scenario.agent(),
+        check_tool_call_args,
+        scenario.succeed(),
+    ],
+)
+```
+
+```typescript [typescript]
+const response = await scenario.run({
+  ...
+  script: [
+    scenario.user("What's the weather in Paris?"),
+    scenario.agent(),
+    (state) => {
+      const toolCalls = state.latestAgentMessage().toolCalls;
+      expect(toolCalls).toBeDefined();
+      expect(toolCalls[0].toolName).toBe("get_current_weather");
+      expect(toolCalls[0].args.location).toBe("Paris");
+    },
+    scenario.succeed(),
+  ],
+});
+```
+:::
+
+## Evaluating Tool Calls with JudgeAgent Criteria
+
+For more nuanced or subjective tool call evaluation, use JudgeAgent with natural language criteria.
+
+:::code-group
+```python [python]
+result = await scenario.run(
+    ...,
+    agents=[
+        WeatherAgent(),
+        scenario.UserSimulatorAgent(),
+        scenario.JudgeAgent(criteria=[
+            "The agent should call the get_current_weather tool with the correct location (Rome) and use the result to answer the user's question."
+        ])
+    ],
+    script=[
+        scenario.user("Should I bring an umbrella to Rome?"),
+        scenario.agent(),
+        scenario.judge(),
+    ],
+)
+```
+
+```typescript [typescript]
+const result = await scenario.run({
+  ...
+  agents: [
+    weatherAgent,
+    scenario.userSimulatorAgent(),
+    scenario.judgeAgent({
+      criteria: [
+        "The agent should call the get_current_weather tool with the correct location (Rome) and use the result to answer the user's question."
+      ],
+    })
+  ],
+  script: [
+    scenario.user("Should I bring an umbrella to Rome?"),
+    scenario.agent(),
+    scenario.judge(), // JudgeAgent will evaluate based on criteria
+    scenario.succeed(),
+  ],
+});
+```
+:::
+
+## Mocking or Scripting Tool Call Results
+
+To make your tests deterministic and avoid backend setup, inject a tool response directly using `scenario.message()`.
+
+:::code-group
+```python [python]
+result = await scenario.run(
+  ...,
+  script=[
+    scenario.user("What's the weather in Paris?"),
+    scenario.agent(),  # Agent calls the tool
```

Comment:
this won't fly, most agent frameworks already call the tool response immediately